File: elixir/pig-latin/lib/pig_latin.ex
defmodule PigLatin do
  @doc """
  Given a `phrase`, translate it a word at a time to Pig Latin.
  """
  @spec translate(phrase :: String.t()) :: String.t()
  def translate(phrase), do:
    phrase |> String.split() |> Enum.map(&translate_word/1) |> Enum.join(" ")

  defp translate_word(word) do
    cond do
      Regex.match?(~r/^[aeiou]|^xr|^yt/, word) -> word <> "ay"
      match_cons_qu = Regex.run(~r/^([^aeiou]*qu)(.*)$/, word) -> rearrange(match_cons_qu)
      match_cons_y = Regex.run(~r/^([^aeiou]+)y(.*)$/, word) -> "y" <> rearrange(match_cons_y)
      match_cons = Regex.run(~r/^([^aeiou]+)(.*)$/, word) -> rearrange(match_cons)
      true -> word
    end
  end

  defp rearrange([_, matched, rest]), do: rest <> matched <> "ay"
end

File: python/pig-latin/pig_latin.py
"""Convert words and phrases to Pig Latin"""

import re

def translate_word(word):
    if re.match(r'[aeiou]|xr|yt', word):
        return word + 'ay'
    if match := re.match(r'([^aeiou]*qu)(.*)', word):
        return match.group(2) + match.group(1) + 'ay'
    if match := re.match(r'([^aeiou]+)(y.*)', word):
        return match.group(2) + match.group(1) + 'ay'
    if match := re.match(r'([^aeiou]+)(.*)', word):
        return match.group(2) + match.group(1) + 'ay'

def translate(text):
    return ' '.join(translate_word(word) for word in text.split(' '))
